graphics/src/ColladaLoader.cc

```diff
@@ -549,6 +554,41 @@ void ColladaLoaderPrivate::LoadNode(tinyxml2::XMLElement *_elem, Mesh *_mesh,
     if (_mesh->HasSkeleton())
       _mesh->MeshSkeleton()->SetNumVertAttached(0);
     this->LoadGeometry(geomXml, transform, _mesh);
+
+    // Associate materials defined in library_materials with the right mesh
+    if (materialEffectMap.size() > 0)
+    {
+      std::string matStr;
+      int matIndex = -1;
+      std::map<std::string, std::string>::iterator iter;
+
+      iter = this->materialEffectMap.find(geomURL);
+      if (iter != this->materialEffectMap.end())
+        matStr = iter->second;
+      MaterialPtr mat = this->LoadMaterial(matStr);
+      matIndex = _mesh->IndexOfMaterial(mat.get());
+      if (matIndex < 0)
+      {
+        matIndex = _mesh->AddMaterial(mat);
+      }
+      if (matIndex < 0)
```

Comment:
Shouldn't this block be combined with the one above?

Reply:
Not really, the first condional override the variable

https://github.com/ignitionrobotics/ign-common/blob/188d06d0ee5d1b9ed3d3afcd2576e268606e8cec/graphics/src/ColladaLoader.cc#L570-L577

Reply:
ah yes!